jetbrains-core/src/software/aws/toolkits/jetbrains/services/ecs/exec/EcsExecUtils.kt

```diff
@@ -95,6 +105,78 @@ object EcsExecUtils : CoroutineScope by ApplicationThreadPoolScope("EcsExec") {
         )
     }
 
+    fun getEc2InstanceTaskRoleArn(project: Project, clusterArn: String, ecsClient: EcsClient, task: software.amazon.awssdk.services.ecs.model.Task): String? {
+        try {
+            val iamClient = project.awsClient<IamClient>()
+            val containerInstanceArn = task.containerInstanceArn()
+            val res = ecsClient.describeContainerInstances(
+                DescribeContainerInstancesRequest
+                    .builder()
+                    .cluster(clusterArn)
+                    .containerInstances(containerInstanceArn).build()
+            )
+            val ec2InstanceId = res.containerInstances().first().ec2InstanceId()
+            val instanceProfileArn = project.awsClient<Ec2Client>().describeInstances(
+                DescribeInstancesRequest.builder().instanceIds(ec2InstanceId).build()
+            ).reservations().first().instances().first().iamInstanceProfile().arn() ?: return null
+            val instanceProfileName = instanceProfileArn.substringAfter(":instance-profile/")
+            return iamClient.getInstanceProfile(GetInstanceProfileRequest.builder().instanceProfileName(instanceProfileName).build()).instanceProfile()
```

Comment:
style in our codebase prefers using the consumer pattern here instead of explicit builders
```kt
iamClient.getInstanceProfile {
    it.instanceProfileName(instanceProfileName)
}
```

Reply:
made the change